File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        left=0
        right=len(people)-1
        people.sort()
        boats=0
        while left<=right:
            if people[left]+people[right]<=limit:
                left+=1
            boats+=1
            right-=1
        return boats

File: 0881-boats-to-save-people/0881-boats-to-save-people.go
func numRescueBoats(people []int, limit int) int {
    sort.Ints(people)
    i,j := 0, len(people) - 1
    ans := 0

    for ; i <= j; j-- {
        ans++
        if (people[i] + people[j] <= limit) { i++ }
    }

    return ans
}